src/WorkspaceSystem/IconGroup.vala

```diff
@@ -111,15 +112,34 @@ public class Dock.IconGroup : Gtk.Box {
             grid.remove (child);
         }
 
+        int icon_size = get_group_icon_size ();
         for (var i = 0; i < int.min (workspace.windows.size, 4); i++) {
             var image = new Gtk.Image.from_gicon (workspace.windows[i].icon) {
-                pixel_size = 16
+                pixel_size = icon_size
             };
 
             grid.attach (image, i % MAX_IN_ROW, i / MAX_IN_COLUMN, 1, 1);
         }
     }
 
+    private int get_group_icon_size () {
+        int icon_size = 8;
+
+        switch (dock_settings.get_int ("icon-size")) {
+            case 64:
+                icon_size = 24;
+                break;
+            case 48:
+                icon_size = 16;
+                break;
+            default:
+                icon_size = 8;
```

Comment:
Probably should handle 32 explicitly since that's a supported size

Maybe we should do something like dock icon size / 3 for default just so we have some kind of attempt at handling in-between sizes?